qds_sdk/resource.py

```diff
@@ -106,6 +107,35 @@ def create(cls, **kwargs):
     def my_element_path(self):
         return self.__class__.element_path(self.id)
 
+    @classmethod
+    def list(cls, page = None, per_page = None):
+        conn = Qubole.agent()
+        url_path = cls.rest_entity_path
+        page_attr = []
+        if page is not None:
+            page_attr.append("page=%s" % page)
+        if per_page is not None:
+            page_attr.append("per_page=%s" % per_page)
+        if page_attr:
+            url_path = "%s?%s" % (cls.rest_entity_path, "&".join(page_attr))
+
+        #Todo Page numbers are thrown away right now
+        resource_json = conn.get(url_path)
+        resource_list = []
+        for s in resource_json[inflection.pluralize(inflection.underscore(cls.__name__))]:
```

Comment:
Because of pagination, the result returned from tapp is of the form: {nezha_data_sources:[], page:1, prev_page:nil, next_page:2}. This line picks up nezha_data_sources by doing this operation.
Since the name of class is NezhaDataSource, first it is convert into plural by adding 's' and then adding '_' before each capital letter, to get nezha_data_sources. And then doing resource_json['nezha_data_sources'] will gives us the result.